src/Template/Element/sentences/audio.ctp

```diff
@@ -25,64 +25,79 @@
  * @link     https://tatoeba.org
  */
 use App\Model\CurrentUser;
+use App\Lib\Licenses;
 
 $hasaudio = count($audios) > 0;
 $shouldDisplayBlock = $hasaudio || CurrentUser::isAdmin();
 if (!$shouldDisplayBlock) {
     return;
 }
 
+$licenseTemplate = <<<EOT
+<a ng-if="vm.getLicenseLink(audio.license)" ng-href="{{vm.getLicenseLink(audio.license)}}">
+  {{vm.getLicenseName(audio.license)}}
+</a>
+<span ng-if="!vm.getLicenseLink(audio.license)">
+  {{vm.getLicenseName(audio.license)}}
+</span>
+EOT;
+
+$audioLicenses = json_encode(Licenses::getAudioLicenses());
+$audiosJson = json_encode($audios);
+
+// Prevent interpolation by AngularJS
+$audioLicenses = str_replace('{{', '\{\{', $audioLicenses);
+$audiosJson = str_replace('{{', '\{\{', $audiosJson);
+
+$this->Html->script('/js/sentences/audio-details.ctrl.js', ['block' => 'scriptBottom']);
+$this->Html->script('/js/directives/audio-button.dir.js', ['block' => 'scriptBottom']);
+$this->AngularTemplate->addTemplate(
+    $this->element('sentence_buttons/audio'),
+    'audio-button-template'
+);
 ?>
-<div class="section md-whiteframe-1dp">
+<div ng-controller="AudioDetailsController as vm"
+     ng-init="vm.init(<?= h($audiosJson) ?>, <?= h($audioLicenses) ?>)"
+     ng-cloak>
+<div ng-if="vm.audios.length > 0"
+     layout="column"
+     class="section audio md-whiteframe-1dp">
     <?php /* @translators: header text in sentence page */ ?>
-    <h2><?php echo __('Audio') ?></h2>
-<?php
+    <h2><?= __n('Audio', 'Audio', count($audios)) ?></h2>
 
-if ($hasaudio) {
-    $this->Audio->displayAudioInfo($audios[0]);
-}
+    <div ng-repeat="audio in vm.audios" ng-class="{'disabled': !audio.enabled}">
+        <h3>
+            <audio-button include-disabled="true" audios="[audio]"></audio-button>
+            <span>
+                <?= format(__('by {username}'), [
+                    'username' => '<a ng-href="{{audio.attribution_url}}">{{audio.author}}</a>'
```

Comment:
If the author is external and doesn't have an `attribution_url`, this results in a "link" pointing nowhere. (I noticed this while trying to change the creator to an invalid username, which to my surprise didn't result in any visual difference. It seems like that is intentionally the way to specify that the recording is from an external source (how do you set the `attribution_url` in that case?) but I think it should be clearer that this is what happened. E.g. to avoid typos in usernames.)

Reply:
You’re right, but this is an existing bug/limitation that I don’t want to address in this pull request.

And you are correct that it is prone to errors ([here is one](https://tatoeba.org/sentences/show/8444414)). There is currently no way to specify external `attribution_url` and `license`. It has been only done once with a [migration script](https://github.com/Tatoeba/tatoeba2/blob/d9fe16a7bdcc610e90f53f764769633075dd01e5/docs/database/updates/2016-12-22_2-data.sql#L79) as part of #1378.